simapp/simd/cmd/genaccounts.go

```diff
@@ -130,38 +131,56 @@ contain valid denominations. Accounts may optionally be supplied with vesting pa
 				return fmt.Errorf("failed to get accounts from any: %w", err)
 			}
 
+			bankGenState := banktypes.GetGenesisStateFromAppState(clientCtx.Codec, appState)
 			if accs.Contains(addr) {
-				return fmt.Errorf("cannot add account at existing address %s", addr)
-			}
+				appendflag, _ := cmd.Flags().GetBool(flagAppendMode)
+				if !appendflag {
+					return fmt.Errorf("cannot add account at existing address %s", addr)
+				}
 
-			// Add the new account to the set of genesis accounts and sanitize the
-			// accounts afterwards.
-			accs = append(accs, genAccount)
-			accs = authtypes.SanitizeGenesisAccounts(accs)
+				genesisB := banktypes.GetGenesisStateFromAppState(clientCtx.Codec, appState)
+				for idx, acc := range genesisB.Balances {
+					if acc.Address != addr.String() {
+						continue
+					}
 
-			genAccs, err := authtypes.PackAccounts(accs)
-			if err != nil {
-				return fmt.Errorf("failed to convert accounts into any's: %w", err)
-			}
-			authGenState.Accounts = genAccs
+					// Create a new coins based on the accounts old coins + new coin appended
+					new_coins, err := sdk.ParseCoinsNormalized(acc.Coins.Add(coins...).String())
+					if err != nil {
```

Comment:
```suggestion
					newCoins, err := sdk.ParseCoinsNormalized(acc.Coins.Add(coins...).String())
					if err != nil {
```

Also, why are we calling `ParseCoinsNormalized`. We're already working on the parsed coins. Just append and sort.

Reply:
Ahh yes, good call I was way overthinking this part, Resolved with 663f8c0e4e87171194d629ee4519ede1323cb7e9